osutil/user.go

```diff
@@ -60,16 +60,19 @@ type AddUserOptions struct {
 // We check the (user)name ourselves, adduser is a bit too
 // strict (i.e. no `.`) - this regexp is in sync with that SSO
 // allows as valid usernames.
+// On systems where there are no adduser, this is the regex that verifies
+// users being created, and serves as a replacement for the regex that adduser
+// was providing.
 //
 // IsValidUsername define what is valid for a "system-user" assertion.
-var IsValidUsername = regexp.MustCompile(`^[a-z0-9][-a-z0-9+._]*$`).MatchString
+var IsValidUsername = regexp.MustCompile(`^[a-z0-9][-a-z0-9._]*$`).MatchString
```

Comment:
I wonder if we should change asserts/system_user.go validSystemUserUsernames to match this now?

Reply:
This is a good point, i'll handle this in a followup